syntax = "proto3";

option java_package = "com.bookmyshow.proto";
option java_outer_classname = "ShowtimeProto";

import 'event.proto';

message Showtime {
  string showtimeId = 1;
  string venueId = 2;
  string movieId = 3;
  string eventId = 4;
  string date = 5;
  string startAt = 6;
  string endAt = 7;
}

message ShowtimeInput {
  string venueId = 1;
  string movieId = 2;
  string eventId = 3;
  string date = 4;
  string startAt = 5;
}

message ShowtimeDetails {
  string showtimeId = 1;
  string startAt = 2;
  string endAt = 3;
}

message ShowtimeEventResponse {
  string venueId = 1;
  Event event = 2;
  string venueName = 3;
  string venueMapUrl = 4;
  string venueLocation = 5;
  ShowtimeDetails showtimes = 6;
}

message ShowtimeMovieResponse {
  string venueId = 1;
  string venueName = 2;
  string venueMapUrl = 3;
  string venueLocation = 4;
  repeated ShowtimeDetails showtimes = 5;
}

message ShowtimeDetailResponse {
  Showtime showtime = 1;

  // movie details
  string movieName = 2;
  string movieDescription = 3;
  string movieImageUrl = 4;
  string moviePosterUrl = 5;
  string movieDuration = 6;
  string movieLanguage = 7;
  string movieGenre = 8;

  // venue details
  string venueName = 9;
  string venueLocation = 10;
  string venueMapUrl = 11;

  // event details
  Event event = 13;

  string movieRating = 12;

}

message ShowtimeMovieResponseList {
  repeated ShowtimeMovieResponse responses = 1;
}

message ShowtimeList {
  repeated Showtime showtimes = 1;
}

message Seats {
  repeated string bookedSeats = 1;
}